### PR TITLE
Use the application bundle ID for keychain work

### PR DESCRIPTION
**Pull request by austinzheng**
Requires https://github.com/realm/realm-object-store/pull/521

### CODE REVIEW

**Comment by austinzheng** [comment]
All failing jobs are CI issues.

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,10 @@ x.x.x Release notes (yyyy-MM-dd)
   service indicates that their credential is no longer valid.
 * Address high CPU usage and hangs in certain cases when processing collection
   notifications in highly-connected object graphs.
+* The keychain service name used by Realm to manage the encryption keys for
```
Comment (bdash) [inline CHANGELOG.md]: "keychain item name" seems like more correct terminology here, and elsewhere in this entry.

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,10 @@ x.x.x Release notes (yyyy-MM-dd)
...
+  sync-related metadata is now set to the bundle identifier. Keys that were
+  previously stored within the Realm-specific keychain service will be
```
Comment (bdash) [inline CHANGELOG.md]: "within a single, shared Realm keychain item"?

**Comment by austinzheng** [comment]
Object store changes I pulled in broke the build, changing status back to 'in progress'...

**Comment by bdash** [comment]
It looks like changes may need to be made to accommodate realm/realm-object-store#512.

**Comment by austinzheng** [comment]
Multiprocess tests are failing on my machine for strange reasons but I wanted to see if CI produced a different result.

**Comment by austinzheng** [comment]
This can be reviewed again. It's nothing more than updating the object store version.

**Comment by austinzheng** [comment]
I fixed this up. It's just a changelog entry and advancing the object store pointer by one commit.

**Pull request merged**